server/worker/worker.go

```diff
@@ -0,0 +1,338 @@
+
+func (api *Client) Post(endpoint string, payload []byte) ([]byte, error) {
+	req, err := api.prepare("POST", endpoint, payload)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &http.Client{}
+	res, err := client.Do(req)
+	if res.StatusCode != http.StatusOK {
+		return nil, errors.New("Request failed with status: " + strconv.Itoa(res.StatusCode))
+	}
+	defer res.Body.Close()
+
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func (api *Client) GetDashboard(uid string) *Dashboard {
+	res, err := api.Get("/api/dashboards/uid/" + uid)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "GetDashboard - api.Get failed",
+			"error", err,
+		)
+
+		return nil
+	}
+
+	var dashboardData map[string]interface{}
+	err = json.Unmarshal(res, &dashboardData)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "GetDashboard - Failed to parse dashboardData response",
+			"error", err,
+		)
+
+		return nil
+	}
+
+	return &Dashboard{
+		Uid:  uid,
+		Data: dashboardData,
+	}
+}
+
+func (api *Client) updateDashboards(dashboards []Dashboard) {
+	for _, dashboard := range dashboards {
+		if dashboard.Title == "test dashboard - empty" {
+			dashboard.Data["overwrite"] = true
+			dashboard.Data["message"] = "PRO - Auto-add analytics panel"
+
+			payload, err := json.Marshal(dashboard.Data)
+			if err != nil {
+				return
+			}
+
+			res, err := api.Post("/api/dashboards/db", payload)
+			if err != nil {
+				level.Info(api.Logger).Log(
+					"status", "error",
+					"message", "updateDashboards - Failed to update dashboard",
+					"error", err,
+				)
+
+				return
+			}
+
+			var responseAsStruct DashboardUpdateResponse
+			err = json.Unmarshal(res, &responseAsStruct)
+			if err != nil {
+				level.Info(api.Logger).Log(
+					"status", "error",
+					"message", "updateDashboards - Failed to parse update response",
+					"error", err,
+				)
+
+				return
+			}
+
+			if responseAsStruct.Status == "success" {
+				level.Info(api.Logger).Log(
+					"status", "success",
+					"message", "Added analytics to "+dashboard.Title,
+					"error", err,
+				)
+			}
+		}
+	}
+}
+
+func (api *Client) getAnalyticsPanelData(panelId int) map[string]interface{} {
+	return map[string]interface{}{
+		"id":    panelId,
+		"title": "Analytics",
+		"type":  "macropower-analytics-panel",
+		"gridPos": map[string]interface{}{
+			"h": 0,
+			"w": 0,
+			"x": 0,
+			"y": 0,
+		},
+		"options": map[string]interface{}{
+			"analyticsOptions": map[string]interface{}{
+				"dashboard":         "$__dashboard",
+				"flatten":           false,
+				"heartbeatAlways":   false,
+				"heartbeatInterval": 60,
+				"postEnd":           false,
+				"postHeartbeat":     false,
+				"postStart":         true,
+				"server":            api.AnalyticsUrl + "/write",
+			},
+		},
+	}
+}
+
+func (api *Client) AddAnalyticsToDashboards() {
```

Comment:
I'd reorder functions according to stepdown rule to make code easier to understand